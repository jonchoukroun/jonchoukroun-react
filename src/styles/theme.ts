export default {
    colors: {
        green: "#008080",
        blue: "#000080",
        black: "#141414",
        lightGray: "#f8f8f8",
        gray: "#c0c0c0",
        darkGray: "#878787",
        white: "#fbfbfb",
    },
    header: {
        fontWeight: "500",
        transform: "scaleX(1.1),",
    },
    body: {
        fontSize: "18px",
    },
    border: {
        line: "2px solid",
        color: "#f8f8f8 #7f787f #7f787f #f8f8f8",
        radius: "0",
    },
    defaultDropShadow: "1px 2px 0px 1px rgba(20, 20, 20, 0.7)",
    darkDropShadow: "1px 1px 0px 3px rgba(0, 0, 0, 0.7)",
};
