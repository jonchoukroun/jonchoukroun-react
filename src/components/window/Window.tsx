import React, { FC, useState } from "react";

import { default as Header } from "./header/Header";
import { default as InsetImage } from "../InsetImage";
import { default as Button } from "../mainButton/MainButton";
import { WindowContainer, ContentContainer } from "./styles";

import styled from "styled-components";

interface IWindowProps {
    readonly title: string;
    readonly shouldMinimize: boolean;
    readonly shouldClose: boolean;
}

const ButtonContainer = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: center;
    flex-grow: 1;
    align-items: center;
`;

const ImageContainer = styled.div`
    margin: 20px auto 40px;
`;

const Window: FC<IWindowProps> = ({ title, shouldMinimize, shouldClose }) => {
    const [imageText, setImageText] = useState("Jon Choukroun | Home");
    const callback = (): Promise<void> => {
        setImageText("Coming soon...");
        return new Promise((res) => res());
    };
    return (
        <WindowContainer>
            <Header
                title={title}
                shouldMinimize={shouldMinimize}
                shouldClose={shouldClose}
            />

            <ContentContainer>
                <ImageContainer>
                    <InsetImage width={640} textOverlay={imageText} />
                </ImageContainer>

                <ButtonContainer>
                    <Button text="OK" action={callback} />
                </ButtonContainer>
            </ContentContainer>
        </WindowContainer>
    );
};

export default Window;
