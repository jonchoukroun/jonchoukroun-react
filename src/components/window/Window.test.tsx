import React from "react";
import Window from "./Window";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import theme from "../../styles/theme";
import { ThemeProvider } from "styled-components";

it("renders successfully", () => {
    const title = "Hey nong man";
    shallow(
        <ThemeProvider theme={theme}>
            <Window title={title} shouldMinimize={true} shouldClose={true} />
        </ThemeProvider>
    );
});

it("renders a header", () => {
    const component = renderer.create(
        <ThemeProvider theme={theme}>
            <Window
                title={"Hey nong man"}
                shouldMinimize={true}
                shouldClose={false}
            />
        </ThemeProvider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
