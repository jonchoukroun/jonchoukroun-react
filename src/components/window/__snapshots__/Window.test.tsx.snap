// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a header 1`] = `
<div
  className="sc-fzozJi cBGTDc"
>
  <div
    className="sc-AxjAm fnUNlU"
  >
    <h4
      className="sc-AxirZ bpGOUS"
    >
      Hey nong man
    </h4>
    <div
      className="sc-AxhCb liSLKZ"
    >
      <button
        className="sc-AxiKw hbMXWK"
      >
        -
      </button>
    </div>
  </div>
  <div
    className="sc-fzoLsD kqoSOU"
  >
    <div
      className="sc-fzplWN dTgfLt"
    >
      <div
        className="sc-AxhUy hLvar"
        style={
          Object {
            "height": "auto",
            "width": 640,
          }
        }
      >
        <svg
          viewBox="0 0 300 24"
        >
          <text
            lengthAdjust="spacing"
            textLength="290"
            x="5"
            y="15"
          >
            Jon Choukroun | Home
          </text>
        </svg>
      </div>
    </div>
    <div
      className="sc-fzpans blSAuB"
    >
      <button
        className="sc-AxgMl hntEic"
        onClick={[Function]}
      >
        <div
          className="sc-Axmtr ijtAQs"
        >
          <span
            style={
              Object {
                "textDecoration": "underline",
              }
            }
          >
            O
          </span>
          K
        </div>
      </button>
    </div>
  </div>
</div>
`;
